presto-test-jdbc-compatibility-old-server/src/test/java/io/prestosql/TestJdbcResultSetCompatibility.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logging;
+import io.prestosql.jdbc.BaseTestJdbcResultSet;
+import io.prestosql.jdbc.PrestoDriver;
+import org.testcontainers.containers.PrestoContainer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true)
+public class TestJdbcResultSetCompatibility
```

Comment:
We now have 2 TestJdbcResultSetCompatibility classes:

```
BaseTestJdbcResultSet (io.prestosql.jdbc)
- TestJdbcResultSetCompatibility (io.prestosql)
- TestJdbcResultSet (io.prestosql.jdbc)
   - TestJdbcResultSetCompatibility (io.prestosql)
```

1. why is the inheritance hierarchy uneven? Maybe it could be flat now? (disregard if it cannot)
2. the `TestJdbcResultSetCompatibility` classes could be renamed to better indicate their specifics, without need to lookup module name (which is long and easy to be stripped out of view)

Reply:
2. done

1. cannot be done without substantial changes to old-driver version of `TestJdbcResultSetCompatibility`. It internally mutates `private TestingPrestoServer server` field declared in `TestJdbcResultSet`.